main.qml

```diff
@@ -357,35 +378,17 @@ ApplicationWindow {
             // hack for webEngineView changing it's background color on crashes
             webView.backgroundColor = "transparent"
 
-            if (webView.tries > 0) {
+            var successfullyLoaded = loadRequest.status == WebEngineView.LoadSucceededStatus;
+            if (successfullyLoaded || webView.tries > 0) {
                 // show the webview if the loading is failing
                 // can fail because of many reasons, including captive portals
                 splashScreen.visible = false
                 pulseOpacity.running = false
             }
 
-            if (loadRequest.status == WebEngineView.LoadSucceededStatus) { 
-                webView.webChannel.registerObject( 'transport', transport );
-
-                // Try-catch to be able to return the error as result, but still throw it in the client context
-                // so it can be caught and reported
-                var injectedJS = "try { initShellComm() } " +
-                        "catch(e) { setTimeout(function() { throw e }); e.message || JSON.stringify(e) }";
-                webView.runJavaScript(injectedJS, function(err) {
-                    splashScreen.visible = false
-                    pulseOpacity.running = false
-
-                    if (!err) {
-                        webView.tries = 0
-                    } else {
-                        errorDialog.text = "Error while applying shell JS." +
-                                " Please consider re-installing Stremio from https://www.stremio.com"
-                        errorDialog.detailedText = err
-                        errorDialog.visible = true
-
-                        console.error(err)
-                    }
-                });
+            if (successfullyLoaded) {
+                removeSplashTimer.running = false;
```

Comment:
theoretically it cannot be executed twice because of this?

Reply:
Yes but only if the UI is loaded before the timer kicks in.